Resources/Post-Processors/Fusion 360/g2core 4th Axis Post.cps

```diff
@@ -48,6 +48,7 @@ properties = {
   useRadius: false, // specifies that arcs should be output using the radius (R word) instead of the I, J, and K words
   useG53: false, // disable to avoid G53 output for safe machine retracts - when disabled you must manually ensure safe retracts
   usePostProcessorSpindleSpeedOverride: false, // Enables Min Max Spindle Overwriting by Fusion 360 Post Processor
+  EnableAAxis:true, // Enable A Axis
```

Comment:
Yeah, that'd work as a toggle but doesn't really help for users of a B axis.

Instead, maybe something like:

```
var aAxis = createAxis({coordinate:0, table:true, axis:[-1, 0, 0], cyclic:true, preference:1});
var bAxis = createAxis({coordinate:0, table:true, axis:[0, -1, 0], cyclic:true, preference:1});
if (properties.EnableAAxis) {
        machineConfiguration = new MachineConfiguration(aAxis);
} else {
        machineConfiguration = new MachineConfiguration(bAxis);
}
```

That's just a rough concept.  I didn't look at the `createAxis()` docs, so the change in that for (attempting to) specify the B axis instead of the A one might not be correct. :smile:

Reply:
If you where to do that you'd need to do 
```
var aAxis = createAxis({coordinate:0, table:true, axis:[-1, 0, 0], cyclic:true, preference:1});
var bAxis = createAxis({coordinate:1, table:true, axis:[0, -1, 0], cyclic:true, preference:1});
if (properties.EnableAAxis) {
        machineConfiguration = new MachineConfiguration(aAxis);
} else {
        machineConfiguration = new MachineConfiguration(bAxis);
}

```
changing the cordinate value from a 0 to a 1 is needed in order for fusion to post B axis code

https://knowledge.autodesk.com/support/autodesk-hsm/learn-explore/caas/sfdcarticles/sfdcarticles/How-to-make-a-3-axis-post-processor-from-the-HSM-post-processor-library-support-4-or-5-axis-motion.html

I'm suggesting that it would instead be separate file as I've been looking through the haas NGC post and it looks like you'd have to first define the machine in Fusion 360s Machine config tab, (which isnt fleshed out)


As for the copyright issue I can completely understand where you are coming from, I'm happy to start up a repository on my own github that you could link to in the wiki, I think autodesk would only get legal if you were trying to sell the post file as that eats into part of their business model of having approved post processor partners, 

Ultimately it's just text on a page that I've modified to do what I want, If I change it enough when does the copyright stop applying, (I'm not trying to turn this thread into a what is copyright, their are thousands of those, haha. we do not need another one)